internal/controller/rbac/provider/roles/roles.go

```diff
@@ -154,14 +155,22 @@ func RenderClusterRoles(pr *v1.ProviderRevision, rs []Resource) []rbacv1.Cluster
 		Rules: withVerbs(rules, verbsEdit),
 	}
 
+	viewRules := rules
+	if pkgmetav1.CapabilitiesContainFuzzyMatch(pr.GetCapabilities(), pkgmetav1.ProviderCapabilitySafeStart) {
+		viewRules = append(viewRules, rbacv1.PolicyRule{
```

Comment:
@jbw976 See here ^^ I am added to the viewRules list. But I don't want these rules to be generally adopted by all sets of rules. So I made this new view list of resources to be used with the `withVerbs` helper.

Reply:
is this safe? should we just make a copy of `rules` instead so that `viewRules` can't be affected by later updates to `rules`? e.g.:
```
viewRules := make([]rbacv1.PolicyRule, len(rules))
copy(viewRules, rules)
```

i'm wondering if when the underlying array that they both share has sufficient capacity so that `append` doesn't need to create a new one, if later updates to `rules` could then conflict and affect `viewRules` 🤔 

It seems like this go playground example shows that could be possible: https://go.dev/play/p/oI1xEqpTHYI 

i think in practice this wouldn't be an issue because `withVerbs()` creates a new slice which then gets assigned to `view.Rules`, but what do you think?

Reply:
good catch. I thought append makes a new slice... fixed